src/kernel/kernel.js

```diff
@@ -86,7 +86,7 @@ export default function buildKernel(kernelEndowments) {
       } else if (kp.state === 'fulfilledToData') {
         const s = `data is not callable, has no method ${msg.method}`;
         // eslint-disable-next-line no-use-before-define
-        reject(msg.kernelPromiseID, makeError(s), []);
+        reject(target.id, makeError(s), []);
```

Comment:
can you tell me about this change? I don't know that we have a specific test for it, but the intention was that sending a message to a promise that was previously fulfilled to data (obviously an error) causes the sender's promise to get rejected. I think `target.id` is the target that's been resolved-to-data, so I don't see how we can reject it after it's already been resolved. I think it should be doing roughly the same as the `kp.state === 'rejected'` case below, except with a static error.

Reply:
Ah, good point, I think it was necessary at some time in the past, but that might have been with a mistaken assumption. I reverted the change.